### PR TITLE
Fix PayPal email field from Vindicia response typo

### PR DESCRIPTION
**Pull request by rondogency**
Vindicia misspells 'payPalEmail' to be 'paypalEmail' in their fetch transaction response, which causes trouble for us to update payment method in our database.

### CODE REVIEW

File: src/ObjectHelper.php [hunk 1]
```diff
@@ -50,6 +50,16 @@ public function buildTransaction(stdClass $object)
             }
         }
 
+        // Vindicia screw up the paypal email field by a camel case typo in the response
+        // we need to deal with "payPalEmail" and "paypalEmail" in case they fix it on their side
+        $paypal_email = null;
```
Comment (nickyr) [inline src/ObjectHelper.php]: should be $paypalEmail

File: src/ObjectHelper.php [hunk 2]
```diff
@@ -50,6 +50,16 @@ public function buildTransaction(stdClass $object)
...
+        if (isset($object->statusLog[0]->payPalStatus->payPalEmail)) {
+            $paypal_email = $object->statusLog[0]->payPalStatus->payPalEmail;
+        }
+        if (isset($object->statusLog[0]->payPalStatus->paypalEmail)) {
```
Comment (nickyr) [inline src/ObjectHelper.php]: this could be an `else if`

File: tests/Message/FetchTransactionRequestTest.php [hunk 3]
```diff
@@ -253,4 +255,107 @@ public function testSendByReferenceFailure()
 
         $this->assertNull($response->getTransaction());
     }
+
+    /**
+     * @return void
+     */
+    public function testSendPayPalSuccess()
+    {
+        $this->setMockSoapResponse('FetchTransactionPayPalSuccess.xml', array(
```
Comment (nickyr) [inline tests/Message/FetchTransactionRequestTest.php]: can you name this FetchPayPalTransactionSuccess? That's the way other things are named

File: tests/Mock/FetchTransactionPayPalFailure.xml [hunk 4]
```diff
@@ -0,0 +1,18 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (nickyr) [inline tests/Mock/FetchTransactionPayPalFailure.xml]: this is the same as FetchTransactionFailure.xml, so we don't need to make this a separate test

File: tests/Mock/FetchTransactionPayPalSuccess.xml [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+          <sortOrder xmlns="" xsi:type="xsd:int">1</sortOrder>
+          <active xmlns="" xsi:type="xsd:boolean">1</active>
+        </sourcePaymentMethod>
+        <statusLog xmlns="" xsi:type="vin:TransactionStatus">
+          <status xmlns="" xsi:type="vin:TransactionStatusType">Authorized</status>
+          <timestamp xmlns="" xsi:type="xsd:dateTime">2018-06-15T08:06:05-07:00</timestamp>
+          <paymentMethodType xmlns="" xsi:type="vin:PaymentMethodType">PayPal</paymentMethodType>
+          <payPalStatus xmlns="" xsi:type="vin:TransactionStatusPayPal">
+            <token xmlns="" xsi:type="xsd:string">[TOKEN]</token>
+            <paypalEmail xmlns="" xsi:type="xsd:string">[PAYPAL_EMAIL]</paypalEmail>
```
Comment (nickyr) [inline tests/Mock/FetchTransactionPayPalSuccess.xml]: This is the spelling they actually return, right?

File: src/ObjectHelper.php [hunk 6]
```diff
@@ -50,6 +50,15 @@ public function buildTransaction(stdClass $object)
             }
         }
 
+        // Vindicia screw up the paypal email field by a camel case typo in the response
+        // we need to deal with "payPalEmail" and "paypalEmail" in case they fix it on their side
```
Comment (nickyr) [inline src/ObjectHelper.php]: can you specify which is documented and which is what they actually return?

File: tests/Mock/FetchTransactionPayPalSuccess.xml (hunk 5 above)
Comment (rondogency) [reply tests/Mock/FetchTransactionPayPalSuccess.xml]: yes

**Comment by rondogency** [comment]
@nickyr fixed those small things you mentioned

File: src/ObjectHelper.php [hunk 7]
```diff
@@ -50,6 +50,16 @@ public function buildTransaction(stdClass $object)
             }
         }
 
+        // Vindicia screw up the paypal email field by a camel case typo
```
Comment (parth21shah) [inline src/ObjectHelper.php]: Can you please reword this to something like
```
There is a discrepancy between Vindicia document and actual Vindicia response regarding 'paypalEmail`` field
```

Comment (nickyr) [reply src/ObjectHelper.php]: and can you say which is which?
